src/repl/scala/tools/nsc/interpreter/MemberHandlers.scala

```diff
@@ -214,27 +214,34 @@ trait MemberHandlers {
     val Import(expr, selectors) = imp
     def targetType = intp.global.rootMirror.getModuleIfDefined("" + expr) match {
       case NoSymbol => intp.typeOfExpression("" + expr)
-      case sym      => sym.thisType
+      case sym      => sym.moduleClass.thisType
     }
-    private def importableTargetMembers = importableMembers(targetType).toList
+    private def importableTargetMembers = importableMembers(exitingTyper(targetType)).toList
```

Comment:
Nope it doesn't introduce new call sites. I just moved `exitingTyper` from both call sites to `importableMembers` itself.